File: solutions/java/darts/1/src/main/java/Darts.java
class Darts {
    int score(double xOfDart, double yOfDart) {
        double d = distance(xOfDart, yOfDart);
        if (d <= 1) {
            return 10;
        } else if (d <= 5) {
            return 5;
        } else if (d <= 10) {
            return 1;
        } else {
            return 0;
        }
    }

    double distance(double x, double y) {
        return Math.sqrt(x * x + y * y);
    }
}

File: solutions/go/darts/1/darts.go
package darts

import "math"

func Score(x, y float64) int {
	d := Distance(x, y)
	switch {
	case d <= 1:
		return 10
	case d <= 5:
		return 5
	case d <= 10:
		return 1
	default:
		return 0
	}
}

func Distance(x, y float64) float64 {
	return math.Sqrt(x*x + y*y)
}
